src/Dfc.CourseDirectory.FindACourseApi/Features/Search/Search.cs

```diff
@@ -74,11 +74,11 @@ public class Handler : IRequestHandler<Query, OneOf<ProblemDetails, SearchViewMo
                 };
             }
 
-            var gotPostcode = !string.IsNullOrWhiteSpace(request.Postcode);
+            Postcode postcode = null;
 
-            if (gotPostcode)
+            if (!string.IsNullOrWhiteSpace(request.Postcode))
```

Comment:
Unnecessary check for null or whitespace, consider combining for readability.

```suggestion
            if (!Postcode.TryParse(request.Postcode, out var postcode))
```

Reply:
It's not unnecessary.

Reply:
How is it necessary? `Postcode.TryParse(...)` checks this?

Reply:
If you insist on keeping it this way it should be combined with the child condition to reduce nesting.

Reply:
It's necessary because postcode parsing (and the subsequent error handling if that fails) should only happen if the request actually specifies a postcode.